src/collections/Pages/components/ViewPageButton.tsx

```diff
@@ -0,0 +1,23 @@
+import { getCanonicalUrlForSlug } from '@/components/Header/utils'
+import { ViewDocumentButton } from '@/components/ViewDocumentButton'
+import { resolveTenant } from '@/utilities/tenancy/resolveTenant'
+import type { BeforeDocumentControlsServerProps } from 'payload'
+
+export const ViewPageButton = async (props: BeforeDocumentControlsServerProps) => {
+  const { id, payload } = props
+  const pageRes = await payload.find({
+    collection: 'pages',
+    limit: 1,
+    pagination: false,
+    depth: 2,
+    where: {
+      id: {
+        equals: id,
+      },
+    },
+  })
+  const page = pageRes.docs[0]
+  const pageTenant = await resolveTenant(page.tenant)
+  const canonicalUrl = (await getCanonicalUrlForSlug(pageTenant.slug, page.slug)) || ''
```

Comment:
In the case where the `await` doesn't throw, why can we get an empty result and have to `|| ''`? Will a `ViewDocumentButton` with an empty URI go to some nice 404/oops page?